### PR TITLE
School 19 create schema part 1

### PR DESCRIPTION
**Pull request by Dusthuynh**

### CODE REVIEW

File: src/shared/enums/common.enum.ts
Comment (tulehuynhnhat) [inline src/shared/enums/common.enum.ts:1]: Should located in its resources

**Review by tulehuynhnhat — DISMISSED** [review]

File: src/shared/enums/common.enum.ts
Comment (Dusthuynh) [reply src/shared/enums/common.enum.ts:1]: Yes, that's right. However, all the Entity have not yet been agreed upon and completed, so I cannot resolve this issue right now. I will resolve it later.

File: src/modules/rooming-subscription-requests/entities/rooming-subscription-request.entity.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import { Column, Entity, ManyToOne } from 'typeorm';
+
+@Entity()
+export class RoomingSubscriptionRequest extends BaseObject {
+	@Column()
+	lessorID: number;
+
+	@Column()
+	roomID: number;
+
```
Comment (huynhhuuloc129) [inline src/modules/rooming-subscription-requests/entities/rooming-subscription-request.entity.ts:13]: Missing "State" column

**Review by huynhhuuloc129 — CHANGES_REQUESTED** [review]

File: src/modules/rooms/entities/room.entity.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import { ROOM_STATE } from 'src/shared/enums/common.enum';
+import { Column, Entity, ManyToOne, OneToMany } from 'typeorm';
+
+@Entity()
+export class Room extends BaseObject {
+	@Column()
+	name: string;
+
+	@Column()
+	roomID: string;
```
Comment (huynhhuuloc129) [inline src/modules/rooms/entities/room.entity.ts]: Remove RoomID

**Review by huynhhuuloc129 — CHANGES_REQUESTED** [review]

File: src/modules/payment-records/entities/payment-record.entity.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+	@Column()
+	monthWaterPrice: number;
+
+	@Column()
+	monthElectricityPrice: number;
+
+	@Column()
+	monthRoomPrice: number;
+
```
Comment (huynhhuuloc129) [inline src/modules/payment-records/entities/payment-record.entity.ts:31]: Add `month, year` column

File: src/modules/rooms/entities/room.entity.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+	@Column()
+	tenantID: string;
+
+	@Column()
+	roomingHouseID: string;
+
```
Comment (huynhhuuloc129) [inline src/modules/rooms/entities/room.entity.ts:19]: Missing `width, height` column

**Review by huynhhuuloc129 — CHANGES_REQUESTED** [review]

File: src/modules/rooming-subscription-requests/entities/rooming-subscription-request.entity.ts (hunk 1 above)
Comment (Dusthuynh) [reply src/modules/rooming-subscription-requests/entities/rooming-subscription-request.entity.ts:12]: line 42

Comment (huynhhuuloc129) [reply src/modules/rooming-subscription-requests/entities/rooming-subscription-request.entity.ts:13]: ?

**Review by tulehuynhnhat — APPROVED** [review]

**Review by huynhhuuloc129 — APPROVED** [review]

**Pull request merged**